### PR TITLE
Merge ROCm installation guide

### PR DESCRIPTION
**Pull request by kmaehashi**
~Depends on #4351~ merged

The main motivation is to prevent dead link as the installation guide is linked from many places.

### CODE REVIEW

File: docs/source/install.rst [hunk 1]
```diff
@@ -1,4 +1,4 @@
-Installation Guide
+Installation
 ==================
```
Comment (leofang) [inline docs/source/install.rst:2]: Nitpick: make this line equally long as the title above
```suggestion
============
```

Comment (kmaehashi) [reply docs/source/install.rst]: :+1:

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by emcastillo** [comment]
CI unrelated

**Pull request merged**